src/main/java/com/atlassian/jira/cloud/jenkins/listeners/InitializePluginListener.java

```diff
@@ -0,0 +1,56 @@
+package com.atlassian.jira.cloud.jenkins.listeners;
+
+import com.atlassian.jira.cloud.jenkins.config.JiraCloudPluginConfig;
+import com.atlassian.jira.cloud.jenkins.config.JiraCloudSiteConfig;
+import com.atlassian.jira.cloud.jenkins.logging.PipelineLogger;
+import com.atlassian.jira.cloud.jenkins.pluginConfigApi.PluginConfigApi;
+import com.atlassian.jira.cloud.jenkins.util.SecretRetriever;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import hudson.Plugin;
+import jenkins.model.Jenkins;
+import okhttp3.OkHttpClient;
+
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import static com.atlassian.jira.cloud.jenkins.util.IpAddressProvider.getIpAddress;
+
+public class InitializePluginListener extends Plugin {
+    private static final Logger LOGGER = Logger.getLogger(InitializePluginListener.class.getName());
+    transient PluginConfigApi pluginConfigApi;
+    transient SecretRetriever secretRetriever;
+
+    public InitializePluginListener() {
+        this.secretRetriever = new SecretRetriever();
```

Comment:
Due to the order things are loaded i cant use magical injections with @inject, manually instantiating instead. Seems to just be a race condition, sometimes works with @inject, but always works manually.

Reply:
e.g. using manual `this.secretRetriever = new SecretRetriever();` instead of a @Inject up top